### PR TITLE
AutoWarmth: fix frontlight toggle during day

### LINKED ISSUE #9728: AutoWarmth plugin "Frontlight off during the day" behavior
Firstly thanks for implementing this feature @zwim.

I started using it and today I noticed something:
- After sunrise, plugin turned my frontlight off while reading (as expected).
- It was still dark in the room so I turned the light back on.
- Then I opened the **File Browser** to open another book
- And frontlight turned off again.

I guess this is not the intended behavior. Because it leaves the user in the dark (literally :)
And it repeats this behavior if I switch between the **Reader view** and **File Browser** again.

**Expected behavior:** Keeping the last light state when opening the **File Browser**.

---

Also a **long-tap decription** of this feature in the UI can be helpful for users.
This is what I am planning to add to the guide, you can use this or edit it:

> This feature turns your frontlight on at sunset and turns it off at sunrise according to the “Currently active parameters” in this plugin. You can override this change by manually turning the frontlight on/off. On the next sunrise/sunset, AutoWarmth will toggle it again if it is necessary.

### PR DESCRIPTION
**Pull request by zwim**
This should fix #9728.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9735)

### CODE REVIEW

File: plugins/autowarmth.koplugin/main.lua [hunk 1]
```diff
@@ -525,6 +561,12 @@ function AutoWarmth:getSubMenuItems()
                     self:updateItems(touchmenu_instance)
                 end
             end,
+            hold_callback = function()
+                UIManager:show(InfoMessage:new{
+                    text = _([[This feature turns your front light on at sunset and off at sunrise according to the “Current Active Parameters” in this plugin.
+You can override this change by manually turning the headlight on/off. At the next sunrise/sunset, AutoWarmth will toggle again if needed.]]),
```
Comment (NiLuJe) [inline plugins/autowarmth.koplugin/main.lua:567]: `headlight`? ;o)

File: plugins/autowarmth.koplugin/main.lua [hunk 2]
```diff
@@ -289,13 +305,34 @@ function AutoWarmth:scheduleMidnightUpdate(from_resume)
...
         -- Schedule the first warmth change
-        self:scheduleNextWarmthChange(now_s, 1, from_resume)
         self:setEventHandlers()
+        self:scheduleNextWarmthChange(now_s, 1, from_resume)
+        -- reset user toggles at sun set or sun rise
+        local sunset_in_s = self.current_times_h[7] * 3600 - now_s
+        if sunset_in_s >= 0 then
+            UIManager:scheduleIn(sunset_in_s, self.setFrontlightState, self, true)
+            local sunrise_in_s = self.current_times_h[5] * 3600 - now_s
+            if sunrise_in_s >=0 then
```
Comment (NiLuJe) [inline plugins/autowarmth.koplugin/main.lua:315]: nit: missing space after comparison operator

File: plugins/autowarmth.koplugin/main.lua [hunk 3]
```diff
@@ -43,11 +43,13 @@ local function frac(x)
     return x - math.floor(x)
 end
 
+-- Local variable that shall survive reloading of the plugin but not a restart
+local fl_turned_off = nil -- true/false if someone (AutoWarmth, gesture ...) has toggled the frontlight
```
Comment (NiLuJe) [inline plugins/autowarmth.koplugin/main.lua:47]: You can leave it in AutoWarmth, as long as you always access it as `AutoWarmth.fl_turned_off`, and never `self.fl_turned_off`.

Comment (zwim) [reply plugins/autowarmth.koplugin/main.lua:47]: I knew I was missing something.

**Comment by zwim** [comment]
Ready to merge?

**Pull request merged**